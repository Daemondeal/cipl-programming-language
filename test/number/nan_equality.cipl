let nan = 0/0

print(nan)
print(nan == 0)
print(nan != 1)

print(nan == nan)
print(nan != nan)