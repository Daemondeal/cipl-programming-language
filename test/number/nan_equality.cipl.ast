(let nan (/ 0 0))
(expression (call print nan))
(expression (call print (== nan 0)))
(expression (call print (!= nan 1)))
(expression (call print (== nan nan)))
(expression (call print (!= nan nan)))

