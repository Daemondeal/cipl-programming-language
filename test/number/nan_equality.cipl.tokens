Token          Lexeme         Line 
LET            let            1    
IDENTIFIER     nan            1    
EQUAL          =              1    
NUMBER         0              1     0
SLASH          /              1    
NUMBER         0              1     0
LINE_END       \n             2    
IDENTIFIER     println        3    
LEFT_PAREN     (              3    
IDENTIFIER     nan            3    
RIGHT_PAREN    )              3    
LINE_END       \n             4    
IDENTIFIER     println        4    
LEFT_PAREN     (              4    
IDENTIFIER     nan            4    
EQUAL_EQUAL    ==             4    
NUMBER         0              4     0
RIGHT_PAREN    )              4    
LINE_END       \n             5    
IDENTIFIER     println        5    
LEFT_PAREN     (              5    
IDENTIFIER     nan            5    
BANG_EQUAL     !=             5    
NUMBER         1              5     1
RIGHT_PAREN    )              5    
LINE_END       \n             6    
IDENTIFIER     println        7    
LEFT_PAREN     (              7    
IDENTIFIER     nan            7    
EQUAL_EQUAL    ==             7    
IDENTIFIER     nan            7    
RIGHT_PAREN    )              7    
LINE_END       \n             8    
IDENTIFIER     println        8    
LEFT_PAREN     (              8    
IDENTIFIER     nan            8    
BANG_EQUAL     !=             8    
IDENTIFIER     nan            8    
RIGHT_PAREN    )              8    
LINE_END       \n             8    
EOF                           8    
