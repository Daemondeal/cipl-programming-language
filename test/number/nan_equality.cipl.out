::stdout::
NaN
false
true
false
true

::stderr::
